Apps/Sandcastle/gallery/Refresh Imagery Provider.html

```diff
@@ -0,0 +1,85 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <meta name="description" content="Use Viewer to start building new applications or easily embed Cesium into existing applications.">
+    <meta name="cesium-sandcastle-labels" content="Beginner, Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+</style>
+
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+
+<video id="trailer" style="display: none;" autoplay loop crossorigin controls>
+    <source src="http://cesiumjs.org/videos/Sandcastle/big-buck-bunny_trailer.webm" type="video/webm">
+    <source src="http://cesiumjs.org/videos/Sandcastle/big-buck-bunny_trailer.mp4" type="video/mp4">
+    <source src="http://cesiumjs.org/videos/Sandcastle/big-buck-bunny_trailer.mov" type="video/quicktime">
+    Your browser does not support the <code>video</code> element.
+</video>
+
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer');
+
+var canvas = document.createElement('canvas');
+canvas.width  = 1;
+canvas.height = 1;
+var ctx = canvas.getContext("2d");
+
+var colorProvider = new Cesium.SingleTileImageryProvider({
+    rectangle: Cesium.Rectangle.fromDegrees(-115.0, 38.0, -107, 39.75),
+    image: canvas
+});
+
+viewer.imageryLayers.addImageryProvider(colorProvider);
+
+var colors = ['red', 'blue', 'green'];
+var index = 0;
+Sandcastle.addDefaultToolbarButton('Change Color', function() {
+    ctx.fillStyle = colors[++index % 3];
+    ctx.fillRect(0, 0, 1, 1);
+
+    colorProvider.reload();
+});
+
+var videoElement = document.getElementById('trailer');
+
+var videoProvider = new Cesium.SingleTileImageryProvider({
+    rectangle: Cesium.Rectangle.fromDegrees(-100.0, 28.0, -80, 38.8),
+    image: videoElement
+});
+
+viewer.imageryLayers.addImageryProvider(videoProvider);
+
+viewer.scene.preRender.addEventListener(function() {
+    videoProvider.reload();
```

Comment:
I think it would be a little weird requiring the the `SingleTileImageryProvider` to take the scene for this purpose. Not sure how else I would do it.